src/heretic/main.py

```diff
@@ -306,8 +319,8 @@ def objective(trial: Trial) -> tuple[float, float]:
             # The value is transformed into the actual min_weight value below.
             min_weight = trial.suggest_float(
                 f"{component}.min_weight",
-                0.0,
-                1.0,
+                min_weight_fraction_range[0],
```

Comment:
:+1: `min_weight_fraction` is a better name than `min_weight` here, we should probably change the variable name itself as well.